glean-core/ios/Glean/Glean.swift

```diff
@@ -263,6 +263,77 @@ public class Glean {
         }
     }
 
+    /// When applications are launched using the custom URL scheme, this helper function will process
+    /// the URL and parse the debug commands
+    ///
+    /// - parameters:
+    ///     * url: A `URL` object containing the Glean debug commands as query parameters
+    ///
+    /// There are 3 available commands that you can use with the Glean SDK debug tools
+    ///
+    /// - `logPings`: If "true", will cause pings that are submitted to also be echoed to the device's log
+    /// - `tagPings`:  This command expects a string to tag the pings with and redirects them to the Glean Debug View
+    /// - `sendPing`: This command expects a string name of a ping to force immediate collection and submission of.
+    ///
+    /// The structure of the custom URL uses the following format:
+    ///
+    /// `<protocol>://glean?<command 1>=<paramter 1>&<command 2>=<parameter 2> ...`
+    ///
+    /// Where:
+    ///
+    /// - `<protocol>://` is the "Url Scheme" that has been added for the app and doesn't matter to Glean.
+    /// - `glean` is required for the Glean SDK to recognize the command is meant for it to process.
+    /// - `?` indicating the beginning of the query.
+    /// - `<command>=<parameter>` are instances of the commands listed above  separated by `&`.
+    ///
+    /// There are a few things to consider when creating the custom URL:
+    ///
+    /// - Invalid commands will trigger an error and be ignored.
+    /// - Not all commands are requred to be encoded in the URL, you can mix and match the commands that you need.
+    /// - Special characters should be properly URL encoded and escaped since this needs to represent a valid URL.
+    public func handleCustomUrl(url: URL) {
+        guard url.host?.removingPercentEncoding == "glean",
+            let components = URLComponents(url: url, resolvingAgainstBaseURL: true),
+            let params = components.queryItems else {
+            logger.error("Error parsing query parameters")
+            return
+        }
+
+        for param in params {
+            switch param.name {
+            case "tagPings":
+                if let pingTag = param.value,
+                    let pingTagPattern = try? NSRegularExpression(pattern: "^[a-zA-Z0-9-]{1,20}$") {
```

Comment:
Can we note this in words somewhere? While the regex is simple, I'm afraid of us losing the context why we do this.

Reply:
That's actually a **_really_** good question.  I went looking for why we match against this particular pattern and couldn't find any restrictions in Glean or the glean_parser that required it.  I couldn't even find anything that looked like it directly required it for the Glean Debug View.  @Dexterp37 or @akkomar can refresh my memory as to why this restriction is in place.

Side note, glean_parser appears to use ping names for tests that [violate the pattern](https://github.com/mozilla/glean_parser/blob/967c6bafd35b66a41c9d2a0e06aaa31d59cb5eaf/tests/test_parser.py#L44).

Reply:
> We reference the pattern in our docs, but we don't provide any explanation or justification as to why.

The [prototype design document](https://docs.google.com/document/d/1w0aJK6AqqGAo_wr-iXMyBsmLYQTMKc5BXuSVkBS-MxU/edit?ts=5c6c335a#heading=h.xo415plei88v) states:

> Behind the scenes, glean will take care of marking the HTTP submissions of tagged pings by using a custom HTTP header, X-Debug-ID. The value for this header is the user-provided tag name, which needs to be constrained to a safe character set: [a-zA-Z0-9-]{1,20}.

This has nothing to do with the ping itself, it's only for our own sanity when communicating within http headers.

Reply:
Thanks!  I had lost the context that that document contained!.  I'll add that statement as both a comment in the code and a line in the docs.